packages/@sanity/schema/src/descriptors/convert.ts

```diff
@@ -632,3 +633,32 @@ function maybeBoolean(val: unknown): boolean | undefined {
   }
   return undefined
 }
+
+function maybeI18n(val: unknown): LocalizedMessage | undefined {
+  if (!isObject(val) || Array.isArray(val)) return undefined
+
+  // Check if it's a valid I18nTextRecord format
+  const validI18n = Object.entries(val).every(([key, value]) => {
+    return (
+      typeof key === 'string' &&
+      isObject(value) &&
+      'key' in value &&
+      'ns' in value &&
+      typeof value.key === 'string' &&
+      typeof value.ns === 'string'
+    )
```

Comment:
If you put this into a function with `is` I believe it's possible to avoid the double-check of `i18nValue` below: https://www.typescriptlang.org/play/?#code/C4TwDgpgBAkgjADgHYFEnAE4igXigbQGdMBLJAcwBooBvAawhAC4piMyqolCW2OBfALoAoYQDMArkgDGwEgHskUEoXjI0mEAAoI6LCyKkK1KXSTyA7kkEBKFrs3LCsRKj3YawqFGmLiBBhBqADcAQwAbCQhBXCgHLC9vDAhgCQwlUEh5MShA3Bw8AHI+CkKoADJyxO9vTIhsqDDI6AKi+QAjACsIWTLKxoioqABCYbwkCXDwiqqamsLAsrIB5pnq+e4lpSahyvXa8HqcnYgAOjzWqGKjcj7Zuag6hpPT7nyiktvhflFJGTlFFAxPJ5Fp2qEMEwAIIYDChEAAHgASj15BgACYIz4mJBmSxIAB8BJstESvm4wDiehIEGceAA8l0esBTvEaYQwRCbABuRIkHJaYZs2ms4IQLBaFRqNyaGwk5KpdKibzAjBQLTk-z4QIhQbRKANYWEEmeOYAejNKyGKi4lkeh3RPjRyVk4RAiR+P2EQA